packages/cli/src/extract.ts

```diff
@@ -138,23 +142,44 @@ export default async function extractAndWrite(
   files: readonly string[],
   opts: ExtractCLIOptions
 ) {
-  const {outFile, ...extractOpts} = opts;
+  const {outFile, throws, ...extractOpts} = opts;
   if (outFile) {
     extractOpts.messagesDir = undefined;
   }
   const extractionResults = await extract(files, extractOpts);
   const printMessagesToStdout = extractOpts.messagesDir == null && !outFile;
-  const extractedMessages = extractionResults
-    .map(m => m.messages)
-    .filter(Boolean)
-    .reduce((all, messages) => [...all, ...messages], []);
+  const extractedMessages = new Map<string, ExtractedMessageDescriptor>();
+  for (const {messages} of extractionResults) {
+    for (const message of messages ?? []) {
+      const {id, description, defaultMessage} = message;
+      if (extractedMessages.has(id)) {
+        const existing = extractedMessages.get(id)!;
+        if (
+          description !== existing.description ||
+          defaultMessage !== existing.defaultMessage
```

Comment:
This should allow for the case where the defaultMessage of one is empty where the other is populated.